src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -59,6 +59,28 @@ public void BuildBasicApplication ([ValueSource (nameof (SupportedTargetFramewor
 			}
 		}
 
+		[Test]
+		[Category ("SmokeTests")]
+		public void SmokeTestBuildWithSpecialCharacters ([Values (false, true)] bool forms)
+		{
+			var testName = "テスト";
```

Comment:
@jonpryor  ^ Given this ... should we even bother?

Reply:
@dellis1972: i think that, to the extent that we *can* bother, we *should* bother.

Rephrased: what can/should be done to improve UX and clarity over what is supported?  Android Studio is, up-front, asserting that only ASCII is supported, and will prevent project creation.  This isn't "ideal", but at least is *up-front* and *clear*.

Visual Studio has no such current check.

Thus, for clarity, we have (at least?) two options:

 1. Update Visual Studio to likewise prevent non-ASCII paths.  *Note*: this might anger certain (important) groups. ;-)
 2. Update *and test* our toolchain so that *either* (a) it works, even though Google doesn't support it, or (b) it doesn't work, and we know specifically *why* it doesn't work, so that we can track it into the future.

Results of (2) may dictate (1).

Yes, even if we get this working, Google may "just break" us in the future, but:

 1. They might not break us!
 2. At least we tried.
 3. In trying, we can provide specific feedback to Google around what breaks/needs fixing.

Reply:
![android_studio_non_ascii](https://user-images.githubusercontent.com/184788/134550831-b5c5335a-11ca-4296-b30c-cb54e3302917.png)

On Unix the error turns into a warning and I can create the project, so I still think it's worth to support this scenario

Reply:
I think as long as we can do it and not regress build times for everyone.